linkerd/proxy/transport/src/addrs.rs

```diff
@@ -15,6 +15,12 @@ pub struct ListenAddr(pub SocketAddr);
 #[derive(Copy, Clone, Debug, Eq, PartialEq, Hash)]
 pub struct ServerAddr(pub SocketAddr);
 
+/// The target address of a connection. This may be the `SO_ORIGINAL_DST` target
+/// address, if present, or the server listener's address if there is no
+/// original destination address.
+#[derive(Copy, Clone, Debug, Eq, PartialEq, Hash)]
+pub struct TargetAddr(pub SocketAddr);
```

Comment:
I don't think we want this... I think we want:

1. `OrigDstAddr` represents the SO_ORIGINAL_DST.
2. `Local<ServerAddr>` represents the server's address.
3. Proxy stacks should. for now take a `Param<Option<OrigDstAddr>>` and return an error when the `OrigDstAddr` is not present -- i.e. we should never do proxy routing based on the `Local<ServerAddr>`.

Reply:
Okay, sounds good --- I thought this PR should preserve the current behavior, but we can change that! :+1: